LSPS2/README.md

```diff
@@ -774,6 +774,23 @@ where the next hop is `jit_channel_scid`:
 
 * SHOULD wait for payment parts of a single payment hash until all
   parts sum up to at least `payment_size_msat`.
+  * SHOULD hold a payment part active for at least 90 seconds
+    (**Rationale** [BOLT 4 Basic Multi-Part Payments Requirements][]
+    requires the final hop to wait at least 60 seconds; this
+    replicates that requirement, and adds 30 seconds for
+    additional LSP-client communication overhead).
+  * SHOULD have a separate such timeout for each part that reaches
+    the LSP.
```

Comment:
I think you want to enforce the timeout on the _first_ part you receive for a scid/payment hash pair? Then, if the timeout expires, fail all parts with the same scid/payment hash pair if the `payment_size_msat` hasn't been reached yet. This should usually fail all parts at the same time. It could be another part arrives after the previous ones have been failed back, in that case that part is held for another 90 seconds, but I think that's OK.

Reply:
BOLT says:

>MUST fail all HTLCs in the HTLC set after some reasonable timeout.

So I think you are right

Reply:
Done.